### ISSUE TITLE
Play audio in background with OpenAL

### ISSUE THREAD

**Issue by drslebedev** [emscripten-core/emscripten#19394]
I have an issue regarding OpenAL and audio playback in the background.

When running using `EM_TIMING_RAF`(`requestAnimationFrame`) as scheduling mode for animation, audio stops playing when tab is inactive and goes to background. When looking into the OpenAL implementation, I found that it explicitly stops the audio [library_openal.js](https://github.com/emscripten-core/emscripten/blob/f711f8c408a084460069b81902290de92e2314bb/src/library_openal.js#LL90C1-L95C8).

One of the workarounds which I consider is to switch from `EM_TIMING_RAF` to `EM_TIMING_SETTIMEOUT` and back when visibility changes.

For this I have my callback method:
```cpp
EM_BOOL em_visibilitychange_callback(int, const EmscriptenVisibilityChangeEvent* e, void*) {
    if (e->hidden) {
        emscripten_set_main_loop_timing(EM_TIMING_SETTIMEOUT, 40);
    } else {  
        emscripten_set_main_loop_timing(EM_TIMING_RAF, 1);
    }
    return EM_TRUE;
}
```

And then set the callback:
```cpp
emscripten_set_visibilitychange_callback(nullptr, false, em_visibilitychange_callback);
```

Is it the right way to "force" `OpenAL` to play in the background?
Are there other options to consider?

**Comment by sbc100** [comment]
The comment says "we need to avoid updating audio altogether when in the background" but doesn't explain further.   Perhaps @juj, who authored that comment in 1595e8cddd2e31fa46ef66c71a55b5833f14f551 can explain more? 

If we want both behaviour perhaps we can give you some kind of runtime toggle or way to monkey-patch the behaviour you want.

**Comment by juj** [comment]
Good point, my comment did not actually specify the root cause for that.

The reason for the above code existing is not to stop the update of audio buffers when in background, since browsers already do that, but the reason is to stop it cleanly.

Firefox was observed to still tick rAF() calls when tab is in background, but do so in geometrically slowing down fashion, i.e. if it would first run the rAF at 16.666msec interval when on foreground, then when the tab was placed to background, it would still call rAF() once after 1 second, then once after 2 seconds, then again once after 4 seconds, then 8, 16, 32, ... and so on. This would result in periodic glitchy audio blips coming out from the speaker whenever the tab went to the background.

So the linked code was authored to prevent rAF()s from leaking in.

I did report that as a bug to Firefox, but I can't find the bugzilla entry of it right now. I don't remember it getting fixed, and I can't remember if Firefox still has that behavior.

The right way to force playback on the background is to use a setTimeout loop or a setInterval loop, where the timeouts are fast enough that the audio buffers will not starve. (e.g. 20ms or 50ms or similar, whatever the needed buffer rate is)
